### PR TITLE
10.0 analytic modules

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: analytic_estimated_amount/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    def button_cancel(self):
+        res = super(PurchaseOrder, self).button_cancel()
+        self._remove_analytic_lines()
+        return res
+
+    @api.multi
+    def _remove_analytic_lines(self):
+        analytic_lines = self.mapped('order_line.analytic_line_id')
+        if analytic_lines:
+            analytic_lines.unlink()
```
Comment (jbchouinard) [inline analytic_estimated_amount/models/purchase_order.py:33]: Not a big deal, but is that necessary, given that _remove_analytic_lines is called on the lines when they are unlinked? so they will be deleted anyway if delete=cascade? (and if delete != cascade, does it make sense to keep the PO lines, but delete the analytic lines, when the PO is deleted?

EDIT: I see now it's called by button_cancel, not just unlink, having the method makes sense, so change my comment to: does it make sense to call it in unlink()?

File: analytic_estimated_amount/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        res = super(SaleOrder, self)._create_analytic_account()
+        self.mapped('order_line')._generate_analytic_line()
+        return res
+
+    @api.multi
+    def _remove_analytic_lines(self):
+        analytic_lines = self.mapped('order_line.analytic_line_id')
+        if analytic_lines:
+            analytic_lines.unlink()
+        return True
```
Comment (jbchouinard) [inline analytic_estimated_amount/models/sale_order.py:47]: Same comment as for PO

**Review by jbchouinard — APPROVED** [review]
2 non-blocking comments. The rest looks good.

File: analytic_estimated_amount/models/purchase_order.py (hunk 1 above)
Comment (ghost) [reply analytic_estimated_amount/models/purchase_order.py:33]: @jbchouinard the relation is many2one from POL to AAL, so no, ondelete='cascade' does not work here. ondelete='cascade' would delete the POL when the AAL is deleted, not the inverse.

File: analytic_estimated_amount/models/sale_order.py (hunk 2 above)
Comment (ghost) [reply analytic_estimated_amount/models/sale_order.py:47]: same thing

**Pull request merged**